src/core/shape_inference/include/experimental_detectron_roi_feature_shape_inference.hpp

```diff
@@ -15,74 +16,66 @@ namespace v6 {
 // outputs:
 //          1.  out_shape = [number_of_ROIs, number_of_channels, output_size, output_size]
 //          2.  out_rois_shape = [number_of_ROIs, 4]
-template <class T>
-void shape_infer(const ExperimentalDetectronROIFeatureExtractor* op,
-                 const std::vector<T>& input_shapes,
-                 std::vector<T>& output_shapes) {
-    using DimType = typename std::iterator_traits<typename T::iterator>::value_type;
+template <class TShape>
+std::vector<TShape> shape_infer(const ExperimentalDetectronROIFeatureExtractor* op,
+                                const std::vector<TShape>& input_shapes,
+                                const ITensorAccessor& tensor_accessor = make_tensor_accessor()) {
+    using TDim = typename TShape::value_type;
+    using namespace ov::util;
+    NODE_VALIDATION_CHECK(op, input_shapes.size() >= 2);
 
-    NODE_VALIDATION_CHECK(op, input_shapes.size() >= 2 && output_shapes.size() == 2);
+    auto output_shapes = std::vector<TShape>();
+    output_shapes.reserve(2);
 
     const auto& rois_shape = input_shapes[0];
-    auto& out_shape = output_shapes[0];
-    auto& out_rois_shape = output_shapes[1];
-
-    // all dimensions is initialized by-default as dynamic
-    out_shape.resize(4);
-    out_rois_shape.resize(2);
-
-    // infer static dimensions
-    out_shape[2] = op->get_attrs().output_size;
-    out_shape[3] = op->get_attrs().output_size;
-    out_rois_shape[1] = 4;
-
-    // infer number_of_ROIs (which may be dynamic/static)
-    auto rois_shape_rank = rois_shape.rank();
+    const auto rois_shape_rank = rois_shape.rank();
     NODE_VALIDATION_CHECK(op, rois_shape_rank.compatible(2), "Input rois rank must be equal to 2.");
 
     if (rois_shape_rank.is_static()) {
+        output_shapes.emplace_back(std::initializer_list<TDim>{rois_shape[0], TDim(dim::inf_bound)});
```

Comment:
The reason to create this dimension here is to avoid create other temporaries for channels number and additional allocation when this channel dimension has to be inserted.
For `StaticShape` this value has no meaning as it will be replaced to correct value or there will be exception. 
For `PartialShape` it will be merged or if all pyramid shapes are dynamic rank then correct value is there.

Reply:
Okay then, thanks